feeds/npm/npm.go

```diff
@@ -72,22 +55,70 @@ func fetchPackages(baseURL string) ([]*Package, error) {
 	if err != nil {
 		return nil, err
 	}
-	return rssResponse.Packages, nil
+	return rssResponse.PackageEvents, nil
 }
 
-// Gets the package version from the NPM.
-func fetchVersionInformation(baseURL, packageName string) (string, error) {
-	resp, err := httpClient.Get(fmt.Sprintf("%s/%s", baseURL, packageName))
+// Gets the package version & corresponding created date from NPM. Returns
+// a slice of {}Package.
+func fetchPackages(baseURL, pkgTitle string, count int) ([]*Package, error) {
+	resp, err := httpClient.Get(fmt.Sprintf("%s/%s", baseURL, pkgTitle))
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	defer resp.Body.Close()
-	v := &PackageVersion{}
-	err = json.NewDecoder(resp.Body).Decode(v)
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	var jsonMap map[string]interface{}
+	err = json.Unmarshal(body, &jsonMap)
 	if err != nil {
-		return "", err
+		return nil, fmt.Errorf("%w : %v for package %s", errJSON, err, pkgTitle)
 	}
-	return v.DistTags.Latest, nil
+
+	// The json string `time` contains versions in date order, oldest to newest.
+	versions, ok := jsonMap["time"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("%w : 'time' not found for package %s ",
+			errJSON, pkgTitle)
+	}
+
+	// If `unpublished` exists in the version map then at a given point in time
+	// the package was 'entirely' removed, the packagEvent(s) received are for package
+	// versions that no longer exist. For a given 24h period no further versions can
+	// be uploaded, with any previous versions never being available again.
+	// https://www.npmjs.com/policies/unpublish
+	_, unPublished := versions["unpublished"]
+
+	if unPublished {
+		unPublishedPackages := []*Package{}
+		for i := 0; i < count; i++ {
+			unPublishedPackages = append(unPublishedPackages, &Package{Unpublished: true})
```

Comment:
Do we not want to include package title information here? Whilst we're not using this data right now we might as well populate the Package struct with the known information.

Reply:
Yeh agreed, no harm in doing so :)